tls/s2n_async_pkey.c

```diff
@@ -262,8 +262,7 @@ S2N_RESULT s2n_async_pkey_sign_async(struct s2n_connection *conn, s2n_signature_
     sign->on_complete = on_complete;
     sign->sig_alg = sig_alg;
 
-    RESULT_GUARD_POSIX(s2n_hash_new(&sign->digest));
-    RESULT_GUARD_POSIX(s2n_hash_copy(&sign->digest, digest));
+    RESULT_GUARD(s2n_hash_new_copy(&sign->digest, digest));
```

Comment:
This is why "s2n_hash_new_copy" is necessary. We can't know what memory to allocate in s2n_hash_new, because we don't know yet which hash implementation we're using. We could theoretically allocate the evp memory in s2n_hash_new and then free it again in s2n_hash_copy when we realize that we're not doing an evp hash, but that's just embarrassingly inefficient.